src/main/msp/msp_build_info.c

```diff
@@ -29,145 +29,141 @@
 #ifdef USE_ACRO_TRAINER
-    options |= MSP_BUILD_OPTION_FLAG_ACRO_TRAINER;
+        BUILD_OPTION_ACRO_TRAINER,
 #endif
 #ifdef USE_AKK_SMARTAUDIO
-    options |= MSP_BUILD_OPTION_FLAG_AKK_SMARTAUDIO;
+        BUILD_OPTION_AKK_SMARTAUDIO,
 #endif
 #ifdef USE_BATTERY_CONTINUE
-    options |= MSP_BUILD_OPTION_FLAG_BATTERY_CONTINUE;
+        BUILD_OPTION_BATTERY_CONTINUE,
 #endif
 #ifdef USE_CAMERA_CONTROL
-    options |= MSP_BUILD_OPTION_FLAG_CAMERA_CONTROL;
+        BUILD_OPTION_CAMERA_CONTROL,
 #endif
 #ifdef USE_DASHBOARD
-    options |= MSP_BUILD_OPTION_FLAG_DASHBOARD;
+        BUILD_OPTION_DASHBOARD,
 #endif
-    sbufWriteU8(dst, options); // byte 3
-    options = 0;
 #ifdef USE_EMFAT_TOOLS
-    options |= MSP_BUILD_OPTION_FLAG_EMFAT_TOOLS;
+        BUILD_OPTION_EMFAT_TOOLS,
 #endif
 #ifdef USE_ESCSERIAL_SIMONK
-    options |= MSP_BUILD_OPTION_FLAG_ESCSERIAL_SIMONK;
+        BUILD_OPTION_ESCSERIAL_SIMONK,
 #endif
 #ifdef USE_FRSKYOSD
-    options |= MSP_BUILD_OPTION_FLAG_FRSKYOSD;
+        BUILD_OPTION_FRSKYOSD,
 #endif
 #ifdef USE_GPS
-    options |= MSP_BUILD_OPTION_FLAG_GPS;
+        BUILD_OPTION_GPS,
 #endif
 #ifdef USE_LED_STRIP
-    options |= MSP_BUILD_OPTION_FLAG_LED_STRIP;
+        BUILD_OPTION_LED_STRIP,
 #endif
 #ifdef USE_LED_STRIP_64
-    options |= MSP_BUILD_OPTION_FLAG_LED_STRIP_64;
+        BUILD_OPTION_LED_STRIP_64,
 #endif
 #ifdef USE_MAG
-    options |= MSP_BUILD_OPTION_FLAG_MAG;
+        BUILD_OPTION_MAG,
 #endif
 #ifdef USE_OSD_SD
-    options |= MSP_BUILD_OPTION_FLAG_OSD_SD;
+        BUILD_OPTION_OSD_SD,
 #endif
-    sbufWriteU8(dst, options); // byte 4
-    options = 0;
 #ifdef USE_OSD_HD
-    options |= MSP_BUILD_OPTION_FLAG_OSD_HD;
+        BUILD_OPTION_OSD_HD,
 #endif
 #ifdef USE_PINIO
-    options |= MSP_BUILD_OPTION_FLAG_PINIO;
+        BUILD_OPTION_PINIO,
 #endif
 #ifdef USE_RACE_PRO
-    options |= MSP_BUILD_OPTION_FLAG_RACE_PRO;
+        BUILD_OPTION_RACE_PRO,
 #endif
 #ifdef USE_SERVOS
-    options |= MSP_BUILD_OPTION_FLAG_SERVOS;
+        BUILD_OPTION_SERVOS,
 #endif
 #ifdef USE_VTX
-    options |= MSP_BUILD_OPTION_FLAG_VTX;
+        BUILD_OPTION_VTX,
 #endif
 #ifdef USE_BRUSHED
-    options |= MSP_BUILD_OPTION_FLAG_BRUSHED;
+        BUILD_OPTION_BRUSHED,
 #endif
 #ifdef USE_DSHOT
-    options |= MSP_BUILD_OPTION_FLAG_DSHOT;
+        BUILD_OPTION_DSHOT,
 #endif
 #ifdef USE_MULTISHOT
-    options |= MSP_BUILD_OPTION_FLAG_MULTISHOT;
+        BUILD_OPTION_MULTISHOT,
 #endif
-    sbufWriteU8(dst, options); // byte 5
-    options = 0;
 #ifdef USE_ONESHOT
-    options |= MSP_BUILD_OPTION_FLAG_ONESHOT;
+        BUILD_OPTION_ONESHOT,
 #endif
 #ifdef USE_PROSHOT
-    options |= MSP_BUILD_OPTION_FLAG_PROSHOT;
+        BUILD_OPTION_PROSHOT,
 #endif
 #ifdef USE_PWM_OUTPUT
-    options |= MSP_BUILD_OPTION_FLAG_PWM_OUTPUT;
-#endif
-    sbufWriteU8(dst, options); // byte 6
+        BUILD_OPTION_PWM_OUTPUT,
+#endif
+    };
+    
+    uint16_t lenOptions = ARRAYLEN(options);
+    for (uint16_t i = 0; i < lenOptions; i++)
```

Comment:
We should not worry about int in loops:
```suggestion
    for (int i = 0; i < lenOptions; i++)
```

Reply:
Also:
```suggestion
    for (unsigned i = 0; i < ARRAYLEN(options); i++)
```

Reply:
> Also:
> ```suggestion
>     for (unsigned i = 0; i < ARRAYLEN(options); i++)
> ```
> 

Please excuse my dumb question, but won't that read the length of the array multiple times as the condition is evaluated on each iteration?